src/commands/dev/edge/setup.rs

```diff
@@ -114,30 +128,32 @@ impl Session {
     }
 }
 
-fn get_session_config(deploy_config: &DeployConfig) -> serde_json::Value {
+fn get_session_config(deploy_config: &HttpRouteDeployConfig) -> serde_json::Value {
     match deploy_config {
-        DeployConfig::Zoned(config) => {
+        HttpRouteDeployConfig::Zoned(config) => {
             let mut routes: Vec<String> = Vec::new();
             for route in &config.routes {
                 routes.push(route.pattern.clone());
             }
             json!({ "routes": routes })
         }
-        DeployConfig::Zoneless(_) => json!({"workers_dev": true}),
+        HttpRouteDeployConfig::Zoneless(_) => json!({"workers_dev": true}),
+        HttpRouteDeployConfig::NoRoutes => unimplemented!(),
     }
 }
 
-fn get_session_address(deploy_config: &DeployConfig) -> String {
+fn get_session_address(deploy_config: &HttpRouteDeployConfig) -> String {
     match deploy_config {
-        DeployConfig::Zoned(config) => format!(
+        HttpRouteDeployConfig::Zoned(config) => format!(
             "https://api.cloudflare.com/client/v4/zones/{}/workers/edge-preview",
             config.zone_id
         ),
         // TODO: zoneless is probably wrong
-        DeployConfig::Zoneless(config) => format!(
+        HttpRouteDeployConfig::Zoneless(config) => format!(
             "https://api.cloudflare.com/client/v4/accounts/{}/workers/subdomain/edge-preview",
             config.account_id
         ),
+        HttpRouteDeployConfig::NoRoutes => unimplemented!(),
```

Comment:
is there a possible implementation of either of these NoRoutes case?

Reply:
not for wrangler dev, since currently it only supports workers that are invokable by a zone route or dotdev subdomain. the unimplemented! here is b/c these two will never be called, as we error out of wrangler dev in the NoRoutes case much earlier in the process.

Reply:
Yeah, it's a little unfortunate to turn what should be compile-time exhaustiveness checks into runtime errors, but I don't have a better suggestion ;)